contracts/modules/staking/StakingPool.sol

```diff
@@ -1240,18 +1253,84 @@ contract StakingPool is IStakingPool, ERC721 {
     return ownerOf(0);
   }
 
-  /* management */
+  /* pool management */
 
-  function addProducts(ProductParams[] memory params) external onlyManager {
-    totalSupply = totalSupply;  // To silence view fn warning. Remove once implemented
-    params;
+  function setProducts(uint[] calldata productsToUpdate, ProductParams[] memory params) external onlyManager {
+    _setProducts(productsToUpdate, params);
   }
 
-  function removeProducts(uint[] memory productIds) external onlyManager {
-    totalSupply = totalSupply;  // To silence view fn warning. Remove once implemented
-    productIds;
+  function _setProducts(uint[] calldata productsToUpdate, ProductParams[] memory params) internal {
+
+    (uint24 globalCapacityRatio, uint16[] memory capacityReductionRatios) = ICover(coverContract).getCapacityRatios(productsToUpdate);
+
+    uint32 _totalTargetWeight = totalTargetWeight;
+    uint32 _totalEffectiveWeight = totalEffectiveWeight;
```

Comment:
Use `uint` for stack/memory variables. It's more gas efficient and prevents from accidental overflows.